From 5143e3ddcf6b7d75d3e382ed1f1e06355b9e7dac Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:04:41 -0400
Subject: [powerpc] Remove the redundant flush_fp_to_thread() in setup_sigcontext()

Message-id: <1410545655-205645-53-git-send-email-dzickus@redhat.com>
Patchwork-id: 94106
O-Subject: [RHEL7 PATCH 052/626] powerpc: Remove the redundant flush_fp_to_thread() in setup_sigcontext()
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 662499d04b6bc73d0ecab0ab876bacd5bbe7d6a7
Author: Kevin Hao <haokexin@gmail.com>
Date:   Sun Jul 14 17:02:04 2013 +0800

    powerpc: Remove the redundant flush_fp_to_thread() in setup_sigcontext()

    In commit c6e6771b(powerpc: Introduce VSX thread_struct and CONFIG_VSX)
    we add a invocation of flush_fp_to_thread() before copying the FPR or
    VSR to users. But we already invoke the flush_fp_to_thread() in this
    function. So remove one of them.

    Signed-off-by: Kevin Hao <haokexin@gmail.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/signal_64.c b/arch/powerpc/kernel/signal_64.c
index a613951..a8e37be 100644
--- a/arch/powerpc/kernel/signal_64.c
+++ b/arch/powerpc/kernel/signal_64.c
@@ -96,8 +96,6 @@ static long setup_sigcontext(struct sigcontext __user *sc, struct pt_regs *regs,
  unsigned long msr = regs->msr;
  long err = 0;
 
- flush_fp_to_thread(current);
-
 #ifdef CONFIG_ALTIVEC
  err |= __put_user(v_regs, &sc->v_regs);
 
-- 
1.7.1